src/main/java/com/customer/service/CustomerServiceImpl.java

```diff
@@ -50,19 +57,56 @@ public Customer createCustomer(Customer customer) {
 	}
 
 	@Override
-	public CustomerResponse findCustomerById(int id) {
-		// TODO Auto-generated method stub
+	public CustomerResponse findCustomerById(int id, Principal principal) {
+		String userName = principal.getName(); // get logged in username
 		Optional<Customer> customer = customerRepository.findById(id);
+
 		if (customer.isPresent()) {
-			CustomerResponse customerResponse = CustomerResponse.builder().id(customer.get().getId())
-					.firstName(customer.get().getFirstName()).lastName(customer.get().getLastName())
-					.joiningDate(customer.get().getJoiningDate()).expiryDate(customer.get().getExpiryDate())
-					.address(customer.get().getAddress()).build();
-			customerResponse.setStatus(customer.get().getJoiningDate(), customer.get().getExpiryDate());
-			return customerResponse;
-		}
-		else
+			if (customer.get().getUserName().equals(userName)) {
+				CustomerResponse customerResponse = CustomerResponse.builder().id(customer.get().getId())
+						.firstName(customer.get().getFirstName()).lastName(customer.get().getLastName())
+						.joiningDate(customer.get().getJoiningDate()).expiryDate(customer.get().getExpiryDate())
+						.address(customer.get().getAddress()).build();
+				customerResponse.setStatus(customer.get().getJoiningDate(), customer.get().getExpiryDate());
+				return customerResponse;
+			} else
+				throw new CustomerNotFoundException("User cann't access the another other details = ", id);
+		} else
 			throw new CustomerNotFoundException("Customer not found for id = ", id);
 	}
-}
 
+	@Override
+	public ResponseEntity<String> updateCustomerDetails(int id, Customer customer, Principal principal) {
+		String userName = principal.getName(); // get logged in username
+		Optional<Customer> customerRes = customerRepository.findById(id);
+		if (customerRes.isPresent()) {
+			if (customerRes.get().getUserName().equals(userName) && customerRes.get().getId() == id) {
+				customer.setId(id);
+				customer.setJoiningDate(customerRes.get().getJoiningDate());
+				customer.setExpiryDate(expiryDate);
+				customer.setPassword(bcryptEncoder.encode(customer.getPassword()));
+				customerRepository.save(customer);
+				return new ResponseEntity<String>("User is updated successfully", HttpStatus.OK);
+			} else
+				throw new CustomerNotFoundException("User cann't access the another user details Id = ", id);
```

Comment:
This shouldnt be called, `CustomerNotFoundException`. It should more be like `Forbidden Exception`